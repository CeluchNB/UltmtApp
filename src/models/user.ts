import { Realm } from '@realm/react'

export const DisplayUserSchema: Realm.ObjectSchema = {
    name: 'DisplayUser',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string',
        localGuest: 'bool?',
    },
}
