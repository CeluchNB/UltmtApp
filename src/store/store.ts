import accountReducer from './reducers/features/account/accountReducer'
import { configureStore } from '@reduxjs/toolkit'
import liveGameReducer from './reducers/features/game/liveGameReducer'
import livePointReducer from './reducers/features/point/livePointReducer'
import managedTeamReducer from './reducers/features/team/managedTeamReducer'

const store = configureStore({
    reducer: {
        account: accountReducer,
        managedTeam: managedTeamReducer,
        liveGame: liveGameReducer,
        livePoint: livePointReducer,
    },
})

export type RootState = ReturnType<typeof store.getState>
export default store
