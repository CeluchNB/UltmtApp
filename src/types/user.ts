import { DisplayTeam } from './team'

export interface CreateUserData {
    firstName: string
    lastName: string
    username: string
    email: string
    password: string
}

export interface DisplayUser {
    _id: string
    firstName: string
    lastName: string
    username: string
}

export interface User extends DisplayUser {
    email: string
    requests: string[]
    playerTeams: DisplayTeam[]
    managerTeams: DisplayTeam[]
    archiveTeams: DisplayTeam[]
    openToRequests: boolean
    private: boolean
    guest: boolean
}

export interface GuestUser {
    _id?: string
    firstName: string
    lastName: string
    username?: string
}
