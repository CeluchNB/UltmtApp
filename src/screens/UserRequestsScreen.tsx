import * as React from 'react'
import * as RequestData from '../services/data/request'
import { AppDispatch } from '../store/store'
import { DetailedRequest } from '../types/request'
import MapSection from '../components/molecules/MapSection'
import PrimaryButton from '../components/atoms/PrimaryButton'
import { Props } from '../types/navigation'
import ScreenTitle from '../components/atoms/ScreenTitle'
import TeamListItem from '../components/atoms/TeamListItem'
import { size } from '../theme/fonts'
import {
    RefreshControl,
    SafeAreaView,
    ScrollView,
    StyleSheet,
    Text,
    View,
} from 'react-native'
import {
    removeRequest,
    selectError,
    selectOpenToRequests,
    selectToggleLoading,
    setOpenToRequests,
} from '../store/reducers/features/account/accountReducer'
import { useColors, useData, useLazyData } from '../hooks'
import { useDispatch, useSelector } from 'react-redux'

const UserRequestsScreen: React.FC<Props> = ({ navigation }) => {
    const { colors } = useColors()
    const dispatch = useDispatch<AppDispatch>()
    const toggleLoading = useSelector(selectToggleLoading)
    const openToRequests = useSelector(selectOpenToRequests)
    const error = useSelector(selectError)

    const [respondRequestId, setRespondRequestId] = React.useState('')
    const [deleteRequestId, setDeleteRequestId] = React.useState('')

    const {
        data: requestData,
        loading,
        error: fetchError,
        refetch,
    } = useData<DetailedRequest[]>(RequestData.getRequestsByUser)

    const { error: deleteRequestError, fetch: callDelete } =
        useLazyData<DetailedRequest>(RequestData.deleteUserRequest)

    const { error: respondRequestError, fetch: callRespond } =
        useLazyData<DetailedRequest>(RequestData.respondToTeamRequest)

    React.useEffect(() => {
        const unsubscribe = navigation.addListener('focus', () => {
            if (!loading) {
                refetch()
            }
        })
        return () => {
            unsubscribe()
        }
    }, [navigation, loading, refetch])

    // Strictly through redux?
    const respondToRequest = async (requestId: string, accept: boolean) => {
        setRespondRequestId(requestId)
        await callRespond(requestId, accept)
        dispatch(removeRequest(requestId))
        refetch()
    }

    // Strictly through redux?
    const deleteRequest = async (requestId: string) => {
        setDeleteRequestId(requestId)
        await callDelete(requestId)
        dispatch(removeRequest(requestId))
        refetch()
    }

    const toggleRosterOpen = () => {
        dispatch(setOpenToRequests({ open: !openToRequests }))
    }

    const styles = StyleSheet.create({
        screen: {
            height: '100%',
            backgroundColor: colors.primary,
        },
        title: {
            alignSelf: 'center',
        },
        error: {
            width: '75%',
            alignSelf: 'center',
            fontSize: size.fontLarge,
            color: colors.gray,
        },
        toggleError: {
            alignSelf: 'center',
            fontSize: size.fontFifteen,
            color: colors.error,
        },
        container: {
            width: '75%',
            alignSelf: 'center',
        },
    })

    if (fetchError) {
        return (
            <SafeAreaView style={styles.screen}>
                <ScrollView
                    refreshControl={
                        <RefreshControl
                            colors={[colors.textSecondary]}
                            refreshing={loading}
                            onRefresh={async () => {
                                refetch()
                            }}
                        />
                    }
                    testID="mt-scroll-view">
                    <ScreenTitle style={styles.title} title="My Requests" />
                    <Text style={styles.error}>{fetchError.message}</Text>
                </ScrollView>
            </SafeAreaView>
        )
    }

    return (
        <SafeAreaView style={styles.screen}>
            <ScrollView
                refreshControl={
                    <RefreshControl
                        onRefresh={async () => {
                            refetch()
                        }}
                        refreshing={loading}
                    />
                }
                testID="mt-scroll-view">
                <ScreenTitle style={styles.title} title="My Requests" />
                <View style={styles.container}>
                    <PrimaryButton
                        text={`${
                            openToRequests ? 'prevent' : 'allow'
                        } requests`}
                        onPress={async () => {
                            toggleRosterOpen()
                        }}
                        loading={toggleLoading}
                        style={styles.title}
                    />
                    {error && error?.length > 0 && (
                        <Text style={styles.toggleError}>{error}</Text>
                    )}
                    <MapSection
                        title="Requests From Teams"
                        showButton={false}
                        showCreateButton={false}
                        listData={
                            requestData?.filter(
                                req => req.requestSource === 'team',
                            ) || []
                        }
                        error={
                            requestData?.filter(
                                req => req.requestSource === 'team',
                            ).length === 0
                                ? 'You are all caught up on requests!'
                                : undefined
                        }
                        renderItem={item => {
                            return (
                                <TeamListItem
                                    key={item._id}
                                    team={
                                        item.teamDetails || {
                                            place: 'Team no',
                                            name: 'longer exists',
                                            teamname: 'deleteme',
                                        }
                                    }
                                    showAccept={true}
                                    showDelete={true}
                                    onAccept={async () => {
                                        await respondToRequest(item._id, true)
                                    }}
                                    onDelete={async () => {
                                        await respondToRequest(item._id, false)
                                    }}
                                    error={
                                        respondRequestError &&
                                        respondRequestId === item._id
                                            ? respondRequestError.message
                                            : undefined
                                    }
                                />
                            )
                        }}
                    />
                    <MapSection
                        title="Requests To Teams"
                        showButton={false}
                        showCreateButton={true}
                        onCreatePress={() => {
                            navigation.navigate('RequestTeam')
                        }}
                        listData={
                            requestData?.filter(
                                req => req.requestSource === 'player',
                            ) || []
                        }
                        error={
                            requestData?.filter(
                                req => req.requestSource === 'player',
                            ).length === 0
                                ? 'You have no open requests!'
                                : undefined
                        }
                        renderItem={item => {
                            return (
                                <TeamListItem
                                    key={item._id}
                                    team={
                                        item.teamDetails || {
                                            place: 'Team no',
                                            name: 'longer exists',
                                            teamname: 'deleteme',
                                        }
                                    }
                                    showAccept={false}
                                    showDelete={true}
                                    onDelete={async () => {
                                        await deleteRequest(item._id)
                                    }}
                                    requestStatus={item.status}
                                    error={
                                        deleteRequestError &&
                                        deleteRequestId === item._id
                                            ? deleteRequestError.message
                                            : undefined
                                    }
                                />
                            )
                        }}
                    />
                </View>
            </ScrollView>
        </SafeAreaView>
    )
}

export default UserRequestsScreen
