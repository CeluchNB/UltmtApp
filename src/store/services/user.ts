import { ApiResponse } from '../../types/services'
import { CreateUserData } from '../../types/user'

export const login = async (
    username: string,
    password: string,
): Promise<ApiResponse> => {
    const response = await fetch('https://ultmt-dev.herokuapp.com/user/login', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            Accept: '*/*',
        },
        body: JSON.stringify({
            email: username,
            password: password,
        }),
    })

    if (response.ok) {
        return { data: await response.json() }
    } else {
        throw new Error(await response.text())
    }
}

export const fetchProfile = async (token: string): Promise<ApiResponse> => {
    const response = await fetch('https://ultmt-dev.herokuapp.com/user/me', {
        method: 'GET',
        headers: {
            Authorization: `Bearer ${token}`,
        },
    })

    if (response.ok) {
        return { data: await response.json() }
    } else {
        throw new Error(await response.text())
    }
}

export const createAccount = async (
    profileData: CreateUserData,
): Promise<ApiResponse> => {
    console.log('making create account call')
    const response = await fetch('https://ultmt-dev.herokuapp.com/user', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            Accept: '*/*',
        },
        body: JSON.stringify({ ...profileData }),
    })

    if (response.ok) {
        return { data: await response.json() }
    } else {
        throw new Error(await response.text())
    }
}
