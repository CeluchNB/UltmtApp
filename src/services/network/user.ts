import { API_URL_V1 } from 'react-native-dotenv'
import { ApiResponse } from '../../types/services'
import { CreateUserData } from '../../types/user'
import EncryptedStorage from 'react-native-encrypted-storage'
import axios, { AxiosResponse } from 'axios'

export const login = async (
    username: string,
    password: string,
): Promise<AxiosResponse> => {
    return await axios.post(`${API_URL_V1}/user/login`, {
        password,
        email: username,
    })
}

export const fetchProfile = async (token: string): Promise<AxiosResponse> => {
    return await axios.get(`${API_URL_V1}/user/me`, {
        headers: {
            Authorization: `Bearer ${token}`,
        },
    })
}

export const createAccount = async (
    profileData: CreateUserData,
): Promise<AxiosResponse> => {
    return await axios.post(`${API_URL_V1}/user`, {
        ...profileData,
    })
}

export const logout = async (token: string): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/user/logout`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
            },
        },
    )
}

export const getLocalToken = async (): Promise<ApiResponse> => {
    try {
        const token = await EncryptedStorage.getItem('jwt_token')
        if (token) {
            return { data: token }
        }
    } catch (error) {
        throw error
    }
    throw new Error('No token available')
}

export const searchUsers = async (term: string): Promise<AxiosResponse> => {
    return await axios.get(`${API_URL_V1}/user/search?q=${term}`)
}

export const leaveTeam = async (
    token: string,
    teamId: string,
): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/user/leave/team?team=${teamId}`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
            },
        },
    )
}

export const getPublicUser = async (id: string): Promise<AxiosResponse> => {
    return await axios.get(`${API_URL_V1}/user/${id}`)
}

export const leaveManagerRole = async (
    token: string,
    teamId: string,
): Promise<AxiosResponse> => {
    return await axios.put(
        `${API_URL_V1}/user/managerLeave?team=${teamId}`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
            },
        },
    )
}

export const requestPasswordRecovery = async (
    email: string,
): Promise<AxiosResponse> => {
    return await axios.post(`${API_URL_V1}/user/requestPasswordRecovery`, {
        email,
    })
}

export const resetPassword = async (
    passcode: string,
    newPassword: string,
): Promise<AxiosResponse> => {
    return await axios.post(`${API_URL_V1}/user/resetPassword`, {
        passcode,
        newPassword,
    })
}

export const setOpenToRequests = async (
    token: string,
    open: boolean,
): Promise<AxiosResponse> => {
    return await axios.put(
        `${API_URL_V1}/user/open?open=${open}`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
            },
        },
    )
}

export const changeName = async (
    token: string,
    newFirstName: string,
    newLastName: string,
): Promise<AxiosResponse> => {
    return await axios.put(
        `${API_URL_V1}/user/changeName`,
        {
            newFirstName,
            newLastName,
        },
        {
            headers: {
                Authorization: `Bearer ${token}`,
            },
        },
    )
}

export const setPrivate = async (
    token: string,
    privateAccount: boolean,
): Promise<AxiosResponse> => {
    return await axios.put(
        `${API_URL_V1}/user/setPrivate?private=${privateAccount}`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
            },
        },
    )
}

export const changeEmail = async (
    email: string,
    password: string,
    newEmail: string,
): Promise<AxiosResponse> => {
    return await axios.put(`${API_URL_V1}/user/changeEmail`, {
        email,
        password,
        newEmail,
    })
}
