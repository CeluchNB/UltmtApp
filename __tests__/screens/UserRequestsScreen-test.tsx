import * as UserData from '../../src/services/data/user'
import { NavigationContainer } from '@react-navigation/native'
import { Props } from '../../src/types/navigation'
import { Provider } from 'react-redux'
import React from 'react'
import UserRequestsScreen from '../../src/screens/UserRequestsScreen'
import { fetchProfileData } from '../../fixtures/data'
import store from '../../src/store/store'
import { fireEvent, render, waitFor } from '@testing-library/react-native'
import {
    setProfile,
    setToken,
} from '../../src/store/reducers/features/account/accountReducer'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

const navigate = jest.fn()
const addListener = jest.fn().mockReturnValue(() => {})
const token = 'token.1234.token'

const props: Props = {
    navigation: {
        navigate,
        addListener,
    } as any,
    route: {} as any,
}

beforeAll(() => {
    store.dispatch(setProfile(fetchProfileData))
    store.dispatch(setToken(token))
})

beforeEach(() => {
    jest.clearAllMocks()
})

it('should match snapshot', async () => {
    const snapshot = render(
        <Provider store={store}>
            <NavigationContainer>
                <UserRequestsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    ).toJSON()

    expect(snapshot).toMatchSnapshot()
})

it('should handle toggle roster status click', async () => {
    const spy = jest
        .spyOn(UserData, 'setOpenToRequests')
        .mockReturnValueOnce(
            Promise.resolve({ ...fetchProfileData, openToRequests: false }),
        )
    const { getByText, queryByText } = render(
        <Provider store={store}>
            <NavigationContainer>
                <UserRequestsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )
    fireEvent.press(getByText('allow requests'))
    await waitFor(() => queryByText('prevent requests'))
    expect(spy).toHaveBeenCalledWith(token, true)
})
