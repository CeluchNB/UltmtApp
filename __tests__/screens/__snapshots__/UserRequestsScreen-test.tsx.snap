// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    testID="mt-scroll-view"
  >
    <RCTRefreshControl />
    <View>
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#ffd600",
              "fontSize": 61.64383561643835,
              "fontWeight": "bold",
            },
            {
              "alignSelf": "center",
            },
          ]
        }
      >
        My Requests
      </Text>
      <View
        style={
          {
            "alignSelf": "center",
            "width": "75%",
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "alignSelf": "center",
              "bottom": undefined,
              "left": undefined,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "top": undefined,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#ffd600",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "minWidth": "auto",
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    false,
                    {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="button"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        {
                          "textAlign": "left",
                        },
                        {
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          {
                            "marginHorizontal": 24,
                            "marginVertical": 10,
                          },
                          {
                            "marginHorizontal": 8,
                          },
                          false,
                          {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    allow requests
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#ffd600",
                  "flex": 1,
                  "fontSize": 61.64383561643835,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Requests From Teams
            </Text>
          </View>
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#ababab",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  place5 name5
                </Text>
                <Text
                  style={
                    {
                      "color": "#ffd600",
                      "fontSize": 30.821917808219176,
                      "fontWeight": "bold",
                    }
                  }
                >
                  @
                  place5name5
                </Text>
                <Text
                  style={
                    {
                      "color": "#ffd600",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  2022
                </Text>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignSelf": undefined,
                  "bottom": undefined,
                  "left": undefined,
                  "position": undefined,
                  "right": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "top": undefined,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": undefined,
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "height": 40,
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                      "width": 40,
                    }
                  }
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "disabled": false,
                      }
                    }
                    accessibilityTraits="button"
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        false,
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="accept-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#40dd50",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰐕
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignSelf": undefined,
                  "bottom": undefined,
                  "left": undefined,
                  "position": undefined,
                  "right": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "top": undefined,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": undefined,
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "height": 40,
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                      "width": 40,
                    }
                  }
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "disabled": false,
                      }
                    }
                    accessibilityTraits="button"
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        false,
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="delete-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#ff0000",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰅖
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#ffd600",
                  "flex": 1,
                  "fontSize": 61.64383561643835,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Requests To Teams
            </Text>
            <View
              collapsable={false}
              style={
                {
                  "alignSelf": undefined,
                  "bottom": undefined,
                  "left": undefined,
                  "position": undefined,
                  "right": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "top": undefined,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "backgroundColor": "#ffd600",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 5,
                      "borderWidth": 0,
                      "display": "flex",
                      "elevation": 0,
                      "flexWrap": "nowrap",
                      "height": 40,
                      "margin": 6,
                      "overflow": "hidden",
                      "width": 40,
                    }
                  }
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "disabled": false,
                      }
                    }
                    accessibilityTraits="button"
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        false,
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 5,
                          },
                        ],
                      ]
                    }
                    testID="create-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#303030",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰐕
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#ababab",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  place6 name6
                </Text>
                <Text
                  style={
                    {
                      "color": "#ffd600",
                      "fontSize": 30.821917808219176,
                      "fontWeight": "bold",
                    }
                  }
                >
                  @
                  place6name6
                </Text>
                <Text
                  style={
                    {
                      "color": "#ffd600",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  2022
                </Text>
                <Text
                  style={
                    {
                      "color": "#ababab",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Pending
                </Text>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignSelf": undefined,
                  "bottom": undefined,
                  "left": undefined,
                  "position": undefined,
                  "right": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "top": undefined,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": undefined,
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "height": 40,
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                      "width": 40,
                    }
                  }
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "disabled": false,
                      }
                    }
                    accessibilityTraits="button"
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        false,
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="delete-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#ff0000",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰅖
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
